h2o-core/src/main/java/hex/ModelMetricsRegressionCoxPH.java

```diff
@@ -0,0 +1,231 @@
+        return true;
+      }
+      if (event1 && time1 < time2) {
+        return true;
+      }
+      if (event2 &&  time2 < time1) {
+        return true;
+      }
+      return false;
+    }
+
+    static class Stats {
+      final long ntotals;
+      final long nNotNaN;
+      final long nconcordant;
+      final long nties;
+
+      Stats() {
+        this(0, 0, 0, 0);
+      }
+
+      Stats(long ntotals, long nNotNaN, long nconcordant, long nties) {
+        this.ntotals = ntotals;
+        this.nNotNaN = nNotNaN;
+        this.nconcordant = nconcordant;
+        this.nties = nties;
+      }
+
+      double c() {
+        return (nconcordant + 0.5d * nties) / ntotals;
+      }
+      
+      long discordant() {
+        return ntotals - nconcordant - nties;
+      }
+
+      Stats plus(Stats s2) {
+        return new Stats(ntotals + s2.ntotals, nNotNaN + s2.nNotNaN, nconcordant + s2.nconcordant, nties + s2.nties);
+      }
+    }
+    
+    static Stats concordance(final Vec startVec, final Vec stopVec, final Vec eventVec, List<Vec> strataVecs, final Vec estimateVec) {
+      final long length = estimateVec.length();
+
+      final Stats stats = concordanceStats(null == startVec ? null : startVec.new Reader(), 
+              stopVec.new Reader(), eventVec.new Reader(),
+              strataVecs.stream().map(it -> it.new Reader()).collect(toList()),
+              estimateVec.new Reader(), length);
+
+      return stats;
+    }
+
+    private static Stats concordanceStats(Vec.Reader startVec, Vec.Reader stopVec, Vec.Reader eventVec, List<Vec.Reader> strataVecs, Vec.Reader estimateVec, long length) {
+      Stream<Integer> allIndexes = IntStream.range(0, (int) length).boxed();
+
+      Map<List<Long>, List<Integer>> byStrata = allIndexes.collect(groupingBy(
+              i -> strataVecs.stream()
+                      .map(v -> v.at8(i))
+                      .collect(toList())
+      ));
+
+      return byStrata.values().stream().map(
+        indexes -> statsForAStrata(startVec, stopVec, eventVec, strataVecs, estimateVec, length, indexes)
+      ).reduce(new Stats(), Stats::plus);
+    }
+
+    private static Stats statsForAStrata(Vec.Reader startVec, Vec.Reader stopVec, Vec.Reader eventVec, List<Vec.Reader> strataVecs, Vec.Reader estimateVec, long length, List<Integer> indexes) {
+      long ntotals = 0;
+      long nNotNaN = 0;
+      long nconcordant = 0;
+      long nties = 0;
+
+      for (final int i : indexes) {
+        for (final int j : indexes) {
+          if (j <= i) { continue; }
+
+          final double t1 = stopVec.at(i) - ((startVec != null) ? startVec.at(i) : 0d);
+          final double t2 = stopVec.at(j) - ((startVec != null) ? startVec.at(j) : 0d);
+
+          final long event1 = eventVec.at8(i);
+          final long event2 = eventVec.at8(j);
+          final double estimate1 = estimateVec.at(i);
+          final double estimate2 = estimateVec.at(j);
+
+          strataVecs.stream().forEach(
```

Comment:
creating streams in a double for loop seems not necessary since this is only evaluated when assertions are enabled

Reply:
done